com.unity.render-pipelines.universal/Runtime/RendererFeatures/DebugCullingFeature.cs

```diff
@@ -0,0 +1,580 @@
+            {
+                if (renderingData.cameraData.camera.cameraType == CameraType.Game && renderingData.cameraData.camera.name != "Preview Camera")
+                {
+                    var camPos = renderingData.cameraData.camera.cameraToWorldMatrix * new Vector4(0,0,0,1);
+                    var camFront = renderingData.cameraData.camera.cameraToWorldMatrix.GetColumn(2);
+                    var shadowCascadesCount = renderingData.shadowData.mainLightShadowCascadesCount;
+
+                    // Test
+                    /*DebugCullingHelpers.DrawPlane(new Vector4(1,0,0,2), 4, Color.red, 10,Color.red );
+                    DebugCullingHelpers.DrawPlane(new Vector4(0,1,0,2), 4, Color.green, 10,Color.green );
+                    DebugCullingHelpers.DrawPlane(new Vector4(0,0,1,2), 4, Color.blue, 10,Color.blue );
+                    DebugCullingHelpers.DrawPlane(new Vector4(1,1,1,3.46f), 4, Color.white, 10,Color.white );*/
+
+                    // Test
+                    /*{
+                        DebugCullingHelpers.DrawPlane(new Vector4(1,0,1,0), 10, Color.green, 5,Color.white );
+
+                        DebugCullingHelpers.DrawBox(new Vector4(0,0,5,1), new Vector3(1, 2, 3 ),  Color.gray );
+                        DebugCullingHelpers.DrawPoint(new Vector4(3,1,0,1), 1,  Color.cyan );
+                    }*/
+
+                    // Origin
+                    if(m_Feature.drawOrigin)
+                    {
+                        Debug.DrawLine( Vector3.zero, new Vector3(1,0,0), Color.red);
+                        Debug.DrawLine( Vector3.zero, new Vector3(0,1,0), Color.green);
+                        Debug.DrawLine( Vector3.zero, new Vector3(0,0,1), Color.blue);
+                    }
+
+                    // Frustum
+                    if (m_Feature.drawCullingFrustum)
+                    {
+                        DebugCullingHelpers.DrawFrustum(renderingData.cameraData.camera.cullingMatrix);
+                        DebugCullingHelpers.DrawAxes(renderingData.cameraData.camera.cameraToWorldMatrix, 0.25f);
+
+                        if (m_Feature.drawCullingFrustumSplits && shadowCascadesCount > 1)
+                        {
+                            var f = renderingData.cameraData.camera.farClipPlane;
+                            var n = renderingData.cameraData.camera.nearClipPlane;
+                            var s = renderingData.cameraData.maxShadowDistance;
+                            var sMax = (s - n) / f;
+                            DebugCullingHelpers.DrawFrustumSplits(renderingData.cameraData.camera.cullingMatrix, sMax, renderingData.shadowData.mainLightShadowCascadesSplit, 0, shadowCascadesCount - 1, Color.gray );
+                        }
+                    }
+
+                    int mainLightIndex = renderingData.lightData.mainLightIndex;
+                    VisibleLight mainLight = renderingData.lightData.visibleLights[mainLightIndex];
```

Comment:
We should check whether a directional light exists (so mainLightIndex != -1) , so that we can view spot lights without a directional light.

Reply:
I fixed this.